### PR TITLE
Drop travis

### LINKED ISSUE #16: Move from Travis to GitHub actions
Unfortunately @TomSmithCGAT, after all your wrangling with Travis we'll need to move to GitHub actions as Travis may no longer be free to use (see [this blog post](https://deanattali.com/blog/migrating-travis-to-github/) amongst others).

Fortunately, this seems easy enough and there are simple instructions in that blog post. I'm happy to do this myself when I get the chance.

### PR DESCRIPTION
**Pull request by csdaw**
Drop Travis CI in favour of Github Actions, resolves #16.

I wasn't sure if Github Actions would work when enabled on a branch, and the answer is it won't! So I think I'll have to merge this with master to see if it actually works or not.

### CODE REVIEW

**Comment by csdaw** [comment]
Nevermind, making the pull request prompted the workflow to begin. Although the Ubuntu checks are failing as MSnbase dependencies aren't installed.

I'll try using a Bioconductor-specific Github Actions workflow instead (see [biocthis](https://github.com/lcolladotor/biocthis)).

**Comment by csdaw** [comment]
Almost ready to go. Only the MacOS check is failing because of a recent update to the `withr` package that has not been added to CRAN yet (see https://cloud.r-project.org/bin/macosx/contrib/4.0/).

Once `withr` is at v2.4.1 then merge this PR.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
...
 ================
 
-
 <!-- badges: start -->
-[![Travis build status](https://travis-ci.org/CambridgeCentreForProteomics/camprotR.svg?branch=master)](https://travis-ci.org/CambridgeCentreForProteomics/camprotR)
-[![codecov](https://codecov.io/gh/CambridgeCentreForProteomics/camprotR/branch/master/graph/badge.svg)](https://codecov.io/gh/CambridgeCentreForProteomics/camprotR)
+
+[![R-CMD-check](https://github.com/CambridgeCentreForProteomics/camprotR/workflows/R-CMD-check/badge.svg)](https://github.com/CambridgeCentreForProteomics/camprotR/actions)
+[![Codecov test
+coverage](https://codecov.io/gh/CambridgeCentreForProteomics/camprotR/branch/master/graph/badge.svg)](https://codecov.io/gh/CambridgeCentreForProteomics/camprotR?branch=master)
```
Comment (TomSmithCGAT) [inline README.md:8]: Should 'Codecov test coverage' all be on one line? Not sure if that matters?

**Review by TomSmithCGAT — COMMENTED** [review]
I can't really review ` .github/workflows/check-bioc.yml` but I understand this is coming from Bioconductor. Is that correct? Aside from adding `cutr`, have you made other significant updates? Anything in particular in this file that needs reviewing?  From my perspective, if it runs the testing OK, all fine with me :+1:

**Comment by TomSmithCGAT** [comment]
Thanks for this Charlotte!

File: README.md (hunk 1 above)
Comment (csdaw) [reply README.md:8]: Yes it should be whoops!

Comment (TomSmithCGAT) [reply README.md:8]: I assume you left that there for me to find so I could at least feel like I contributed something :wink:

**Comment by csdaw** [comment]
I generated `.github/workflows/check-bioc.yml` using [biocthis](https://github.com/lcolladotor/biocthis) and then modified it slightly. Technically `biocthis` is not an official Bioconductor core package. However it was the easiest way I found to implement GitHub actions CI with a package like camprotR that Imports Bioconductor packages.

**Comment by csdaw** [comment]
> I can't really review ` .github/workflows/check-bioc.yml` but I understand this is coming from Bioconductor. Is that correct? Aside from adding `cutr`, have you made other significant updates? Anything in particular in this file that needs reviewing? From my perspective, if it runs the testing OK, all fine with me 👍

No other significant changes! I just want to get this into master and then I'll pull it into the `{ts}_test_test_test` branch and start writing more tests.

**Comment by TomSmithCGAT** [comment]
Can't you merge this branch into `{ts}_test_test_test` directly in that case. Or am I missing something?

**Comment by csdaw** [comment]
Yes you can. Good point I hadn't thought of that 😅 . I think I can edit the PR to change the target branch?

**Comment by TomSmithCGAT** [comment]
Great. So just waiting on `withr` update then?

**Comment by csdaw** [comment]
Nope that's already solved otherwise the MacOS check wouldn't pass. I'll merge now 😁.

**Pull request merged**